components/provisioner/internal/provisioning/input_converter.go

```diff
@@ -137,6 +177,24 @@ func (c converter) providerSpecificConfigFromInput(input *gqlschema.ProviderSpec
 	return nil, apperrors.BadRequest("provider config not specified")
 }
 
+func (c converter) providerSpecificConfigFromUpgradeInput(input *gqlschema.ProviderSpecificInput) (model.GardenerProviderConfig, error) {
```

Comment:
This function is exactly the same as `providerSpecificConfigFromInput` and I do not think it is used.